dask_geopandas/io/arrow.py

```diff
@@ -0,0 +1,414 @@
+import copy
+import json
+import math
+from typing import TYPE_CHECKING
+
+from dask.base import compute_as_if_collection, tokenize
+from dask.dataframe.core import new_dd_object, Scalar
+from dask.highlevelgraph import HighLevelGraph
+from dask.layers import DataFrameIOLayer
+from dask.utils import natural_sort_key, apply
+
+import pandas as pd
+import geopandas
+import shapely.geometry
+
+from fsspec.core import get_fs_token_paths
+
+
+if TYPE_CHECKING:
+    import pyarrow
+
+
+def _update_meta_to_geodataframe(meta, schema_metadata):
+    """
+    Convert meta to a GeoDataFrame and update with potential GEO metadata
+    """
+    if schema_metadata and b"geo" in schema_metadata:
+        geo_meta = json.loads(schema_metadata[b"geo"])
+        geometry_column_name = geo_meta["primary_column"]
+        crs = geo_meta["columns"][geometry_column_name]["crs"]
+        geometry_columns = geo_meta["columns"]
+    else:
+        # TODO we could allow the user to pass those explicitly if not
+        # stored in the metadata
+        raise ValueError(
+            "Missing geo metadata in the Parquet/Feather file. "
+            "Use dask.dataframe.read_parquet/read_feather() instead."
```

Comment:
`dask.dataframe.read_feather()` doesn't exist (ref https://github.com/dask/dask/issues/6865) so we shouldn't point to it.